smarts/core/utils/traffic_history_serivce.py

```diff
@@ -77,6 +82,20 @@ def __init__(self, history_file_path):
         self._prepare_next_batch()
         self._receive_data_conn.recv()
 
+    def teardown(self):
+        if self.is_in_use:
+            self._request_queue.put(Traffic_history_service.QueueDone())
```

Comment:
nit: Not very important in this case but I have had this thought that it might be useful to drain the queue before adding the `QueueDone` signal.

Reply:
while I was trying to drain the queue before adding `QueueDone`, I noticed the check for QueueDone was not reached in the child process.
```
        while True:
            print("start")
            historyRange = request_queue.get()
            print("got: ",historyRange)
            if type(historyRange) is Traffic_history_service.QueueDone:
                print("Found done")
                break
```
In fact, `print("got: ",historyRange)` was not reached either, then I tried putting other values in the queue and `print("got: ",historyRange)` also didn't show up. Though `teardown` method was reached. Is this related to putting items in a queue in `___del___` method?

Reply:
The reason this happens is `__del__` for traffic_history_service happens even after smarts calls destroy, when teardown is called, the child process `self._fetch_history_proc` is no longer alive.....